internal/pkg/template/overrides.go

```diff
@@ -43,19 +44,37 @@ type CFNResource struct {
 	LogicalID string
 }
 
+// CDKImport is the interface to import a CDK package.
+type CDKImport interface {
+	ImportName() string
+	ImportShortRename() string
+}
+
 type cfnResources []CFNResource
 
-// UniqueTypes returns the list of unique CFN types.
-func (rs cfnResources) UniqueTypes() []CFNType {
-	var uniqueTypes []CFNType
-	seen := make(map[CFNType]struct{})
+// Imports returns a list of CDK imports for a given list of CloudFormation resources.
+func (rs cfnResources) Imports() []CDKImport {
+	// Find a unique CFN type per service.
+	// For example, given "AWS::ECS::Service" and "AWS::ECS::TaskDef" we want to retain only one of them.
+	seen := make(map[string]CFNType)
 	for _, r := range rs {
-		if _, ok := seen[r.Type]; !ok {
-			uniqueTypes = append(uniqueTypes, r.Type)
+		serviceName := strings.Split(strings.ToLower(string(r.Type)), "::")[1]
```

Comment:
panic if invalid type? should we check length of split?

Reply:
For visibility, I made this assumption everything in this file :lolsob: because of this:
> Resource type identifiers **always** take the following form:
> `service-provider::service-name::data-type-name`
> https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html

I know our `Custom::` types are not of that format, but we never pass them down here (although yea it makes the code a bit more fragile)